airbyte-integrations/connectors/destination-snowflake/src/main/kotlin/io/airbyte/integrations/destination/snowflake/client/SnowflakeAirbyteClient.kt

```diff
@@ -190,25 +198,60 @@ class SnowflakeAirbyteClient(
         if (snowflakeConfiguration.legacyRawTablesOnly) {
             return
         }
-        val columnsInDb = getColumnsFromDb(tableName)
-        val columnsInStream = getColumnsFromStream(stream, columnNameMapping)
-        val (addedColumns, deletedColumns, modifiedColumns) =
-            generateSchemaChanges(columnsInDb, columnsInStream)
+        super.ensureSchemaMatches(stream, tableName, columnNameMapping)
+    }
+
+    override suspend fun discoverSchema(tableName: TableName): Pair<TableSchema, Unit> {
+        return Pair(
+            TableSchema(getColumnsFromDb(tableName)),
+            Unit,
+        )
+    }
+
+    override fun computeSchema(
+        stream: DestinationStream,
+        columnNameMapping: ColumnNameMapping
+    ): Pair<TableSchema, Unit> {
+        return Pair(
+            TableSchema(getColumnsFromStream(stream, columnNameMapping)),
+            Unit,
+        )
+    }
 
+    override fun diff(actualSchemaInfo: Unit, expectedSchemaInfo: Unit) {
```

Comment:
Why are the parameters marked as `Unit`?  Is this to get around some generic issue with Kotlin and what we have type-wise in Snowflake?

Reply:
sort of. it's more of a hack to support certain destinations that need to pass around more info than just column+type - https://github.com/airbytehq/airbyte/pull/69090#discussion_r2482942701